uml/__init__.py

```diff
@@ -66,7 +66,8 @@ def behavior_add_parameter(self, name: str, param_type: str, direction: str, opt
     :param optional: True if the Parameter is optional; default is False
     :return: Parameter that has been added
     """
-    param = Parameter(name=name, type=param_type, direction=direction, is_ordered=True, is_unique=True)
+    idx = len(self.parameters)
+    param = OrderedPropertyValue(index=idx, property_value=Parameter(name=name, type=param_type, direction=direction, is_ordered=True, is_unique=True))
     self.parameters.append(param)
     param.upper_value = literal(1)  # all parameters are assumed to have cardinality [0..1] or 1 for now
```

Comment:
We're losing the upper and lower bound information because it's being set on the ordered property value rather than the parameter.